go-controller/pkg/ovn/gateway_init.go

```diff
@@ -21,6 +25,67 @@ import (
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
+// cleanupStalePodSNATs removes SNATs against nodeIP for the given node if the SNAT.logicalIP isn't an active podIP on this node
+// We don't have to worry about missing SNATs that should be added because addLogicalPort takes care of this for all pods
+// when RequestRetryObjs is called for each node add.
+// This is executed only when disableSNATMultipleGateways = true
+//
+// NOTE: On startup libovsdb adds back all the pods and this should normally update all existing SNATs
+// accordingly. Due to a stale egressIP cache bug https://issues.redhat.com/browse/OCPBUGS-1520 we ended up adding
+// wrong pod->nodeSNATs which won't get cleared up unless explicitly deleted.
+// NOTE2: egressIP SNATs are synced in EIP controller.
+func (oc *Controller) cleanupStalePodSNATs(nodeName string, nodeIPs []*net.IPNet) error {
+	if !config.Gateway.DisableSNATMultipleGWs {
+		return nil
+	}
+	options := metav1.ListOptions{
+		FieldSelector:   fields.OneTermEqualSelector("spec.nodeName", nodeName).String(),
+		ResourceVersion: "0",
+	}
+	pods, err := oc.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), options)
```

Comment:
TODO: In the next iteration - maybe just use watchFactory to fetch this, need to see if I can list for a specific node and provide that as filter into GetPods.

Reply:
Couldn't get to this for now, the watchfatory lister doesn't really take fieldSelectors as input. Will investigate more and do FUP